test/util/check-deps.sh

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env bash
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/29015#discussion_r1599843905

> I think rather than `test/` this should go into `contrib/devtools`, which contains functionally similar scripts like `symbol-check`, or `security-check`.

Good suggestion, it does fit in more with those tools, moved over there.